upup/pkg/fi/cloudup/awstasks/natgateway.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awstasks
+
+import (
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/golang/glog"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/awsup"
+	//"k8s.io/kops/upup/pkg/fi/cloudup/terraform"
+	"fmt"
+)
+
+//go:generate fitask -type=NatGateway
+type NatGateway struct {
+	Name      *string
+	ElasticIp *ElasticIP
+	Subnet    *Subnet
+	ID        *string
+}
+
+var _ fi.CompareWithID = &NatGateway{} // Validate the IDs
+
+func (e *NatGateway) CompareWithID() *string {
+	return e.ID
+}
+
+func (e *NatGateway) Find(c *fi.Context) (*NatGateway, error) {
```

Comment:
This is the code to look for an existing NAT Gateway in AWS. This code is roughly on par with the other Find() methods in `awstask`